### PR TITLE
Feat/user-page

### PR DESCRIPTION
**Pull request by zakrzaq**
- [x] reservation service with auth
- [x] user page
- [x] navigation tabs
- [x] user detail form
- [x] api integrated
- [x] reservation interfaces
- [x] reservations list UI

### CODE REVIEW

File: src/stores/user.ts [hunk 1]
```diff
@@ -82,9 +90,9 @@ export const useUserStore = defineStore("userStore", {
       }
     },
     async getUser() {
-      this.callLogout();
       try {
         const response = await getUser();
+        console.log(response);
```
Comment (szymonlicau) [inline src/stores/user.ts:95]: I've seen several console.logs, are we sure we want them here? 🤔

**Review by szymonlicau — CHANGES_REQUESTED** [review]

File: src/api/reservationsService.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export const getReservations = async (
+  email: string,
+  page = 1,
+  per_page = 10
+) => {
+  const response = await defaultClient.get(
+    `/reservations?user_email=${email}&page=${page}&per_page=${per_page}`,
+    {
+      headers: { Authorization: `Bearer ${userStore.authToken}` },
+    }
```
[the remark this answers is not in the document]
Comment (zakrzaq) [reply src/api/reservationsService.ts:15]: Everyday is a school day :) I will definitely use this approach - look so much more structured and neat!

File: src/components/user/UserReservations.vue [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import { useUserStore } from "@/stores/user";
+import BaseCard from "@/components/base/BaseCard.vue";
+import ReservationItem from "@/components/reservations/ReservationItem.vue";
+import LoadingSpinner from "@/components/LoadingSpinner.vue";
+import NoResults from "@/components/NoResults.vue";
+import type { ShowTicket } from "@/types/reservations";
+const userStore = useUserStore();
+
+const userReservations = ref();
+const ticketList = ref<[] | ShowTicket[]>([]);
```
[the remark this answers is not in the document]
Comment (zakrzaq) [reply src/components/user/UserReservations.vue:13]: Nice one, I think I sometime over do it and overthink it when working with TS...

File: src/components/user/UserReservations.vue [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  <template v-else>
+    <div v-if="ticketList" class="user-reservations">
+      <BaseCard>
+        <h3 class="user-reservations__heading">Upcoming</h3>
+        <template v-if="upcomingTickets">
+          <ReservationItem
+            v-for="ticket in upcomingTickets"
+            :key="ticket.id"
+            :ticket="ticket"
+          ></ReservationItem>
```
[the remark this answers is not in the document]
Comment (zakrzaq) [reply src/components/user/UserReservations.vue:63]: 😅 This one managed to get snuck past me

File: src/stores/user.ts (hunk 1 above)
Comment (zakrzaq) [reply src/stores/user.ts:95]: No excuse here, need to improve me clean up game :)

**Review by szymonlicau — APPROVED** [review]
Great work overall 😄 Seems like you are slowly getting TypeScript nailed down ❤️

**Pull request merged**